### ISSUE TITLE
Add option to filter licenses by targets

### ISSUE THREAD

**Issue by jwodder** [sstadick/cargo-bundle-licenses#19]
If one is planning to only build & distribute binaries of a project for a limited number of targets, it doesn't make much sense to bundle licenses for dependencies exclusive to other targets that will never be used.  (For example, if a project depends on `chrono` and its `clock` feature, then `iana-time-zone` will be pulled into the dependency tree, and this package depends on `android_system_properties` on Android and `js-sys` & `wasm-bindgen` on WASM, but if you're only building for Linux, macOS, and Windows, you shouldn't need to bundle those packages' licenses.)

I therefore request a feature for specifying a set of target triples by which to filter the dependencies for which licenses are bundled.

**Comment by sstadick** [comment]
Agreed, this would be nice to have. I'm not sure how to implement this off the top of my head as this tool heavily relies on cargo-metadata to build up the deps, and I'm not sure that is target aware.

**Comment by jwodder** [comment]
`cargo metadata` has a `--filter-platform <triple>` option for only showing dependencies for a given triple, so one way to implement this would be run the command for each triple and combine the results.

**Comment by jwodder** [comment]
I've just discovered the [`krates`](https://github.com/EmbarkStudios/krates) crate, which provides functionality for determining all dependencies of a package filtered by [kind](https://docs.rs/krates/latest/krates/struct.Builder.html#method.ignore_kind) and/or [target](https://docs.rs/krates/latest/krates/struct.Builder.html#method.include_targets), among other things.

**Comment by sstadick** [comment]
That does seem like it should work...

I don't have a lot of spare time at the moment but I do want to work this in at a future point!
